### PR TITLE
Make sure mirror registry is ready

### PR DESCRIPTION
**Pull request by yunjiang29**

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-private-p1-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-private-p1-f28-destructive

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-private-p1-f28-destructive

**Comment by yunjiang29** [comment]
result https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/40793/rehearse-40793-periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-aws-ipi-disconnected-private-p1-f28-destructive/1674343372568924160/artifacts/aws-ipi-disconnected-private-p1-f28-destructive/cucushift-installer-check-mirror-registry-services/build-log.txt
```
+ [[ ! -e /tmp/secret/bastion_public_address ]]
+ BASTION_PUBLIC_ADDRESS=ec2-54-198-167-192.compute-1.amazonaws.com
+ SSH_PRIV_KEY_PATH=/var/run/secrets/ci.openshift.io/cluster-profile/ssh-privatekey
+ BASTION_SSH_USER=core
+ check_mirror_registry_response
+ local try=0
+ local max_retries=20
+ local interval=30
+ ((  try < max_retries  ))
+ echo 'Checking mirror registry response code 0/20'
Checking mirror registry response code 0/20
++ curl -o /dev/null -I -k -s -w '%{http_code}' https://ec2-54-198-167-192.compute-1.amazonaws.com:5000
+ http_code=200
+ [[ 200 != \2\0\0 ]]
+ echo Succeed.
Succeed.
+ return 0
+ check_mirror_registry_service_status
+ local try=0
+ local max_retries=20
+ local interval=30
+ ((  try < max_retries  ))
+ echo 'Checking mirror registry service status 0/20'
Checking mirror registry service status 0/20
+ ssh -o UserKnownHostsFile=/dev/null -o IdentityFile=/var/run/secrets/ci.openshift.io/cluster-profile/ssh-privatekey -o StrictHostKeyChecking=no core@ec2-54-198-167-192.compute-1.amazonaws.com 'sudo systemctl status poc-registry-5000.service'
Could not create directory '/.ssh'.
Warning: Permanently added 'ec2-54-198-167-192.compute-1.amazonaws.com,54.198.167.192' (ECDSA) to the list of known hosts.
+ echo 'Debug: content of /logs/artifacts/poc-registry-5000.status'
Debug: content of /logs/artifacts/poc-registry-5000.status
+ cat /logs/artifacts/poc-registry-5000.status
● poc-registry-5000.service - OpenShift POC HTTP for PXE Config
     Loaded: loaded (/etc/systemd/system/poc-registry-5000.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-06-29 09:18:54 UTC; 4min 12s ago
    Process: 1092 ExecStartPre=/usr/bin/podman rm poc-registry-5000 (code=exited, status=1/FAILURE)
    Process: 1225 ExecStartPre=/usr/bin/chcon -Rt container_file_t /opt/registry-5000 (code=exited, status=0/SUCCESS)
   Main PID: 1226 (podman)
      Tasks: 11 (limit: 4597)
     Memory: 32.5M
        CPU: 1.451s
     CGroup: /system.slice/poc-registry-5000.service
             ├─1226 /usr/bin/podman run --name poc-registry-5000 -p 5000:5000 --net host -v /opt/registry-5000/data:/var/lib/registry:z -v /opt/registry-5000/auth:/auth -v /opt/registry-5000/certs:/certs:z -v /opt/registry-5000/config.yaml:/etc/docker/registry/config.yml quay.io/openshifttest/registry:2
             └─1554 /usr/bin/conmon --api-version 1 -c 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175 -u 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175 -r /usr/bin/crun -b /var/lib/containers/storage/overlay-containers/0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175/userdata -p /run/containers/storage/overlay-containers/0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175/userdata/pidfile -n poc-registry-5000 --exit-dir /run/libpod/exits --full-attach -s -l k8s-file:/var/lib/containers/storage/overlay-containers/0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175/userdata/ctr.log --log-level warning --runtime-arg --log-format=json --runtime-arg --log --runtime-arg=/run/containers/storage/overlay-containers/0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175/userdata/oci-log --conmon-pidfile /run/containers/storage/overlay-containers/0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /var/lib/containers/storage --exit-command-arg --runroot --exit-command-arg /run/containers/storage --exit-command-arg --log-level --exit-command-arg warning --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/libpod --exit-command-arg --runtime --exit-command-arg crun --exit-command-arg --storage-driver --exit-command-arg overlay --exit-command-arg --storage-opt --exit-command-arg overlay.mountopt=nodev,metacopy=on --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg container --exit-command-arg cleanup --exit-command-arg 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175

Jun 29 09:19:00 ip-10-0-5-65 podman[1226]: 2023-06-29 09:19:00.540484211 +0000 UTC m=+6.078481382 container create 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175 (image=quay.io/openshifttest/registry:2, name=poc-registry-5000)
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: 2023-06-29 09:19:01.165028885 +0000 UTC m=+6.703025962 container init 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175 (image=quay.io/openshifttest/registry:2, name=poc-registry-5000)
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: 2023-06-29 09:19:01.242480588 +0000 UTC m=+6.780477659 container start 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175 (image=quay.io/openshifttest/registry:2, name=poc-registry-5000)
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: 2023-06-29 09:19:01.242832722 +0000 UTC m=+6.780829814 container attach 0a1345c2376bffcfea12c9422817df642214c98aa50259923abc3cab34727175 (image=quay.io/openshifttest/registry:2, name=poc-registry-5000)
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: time="2023-06-29T09:19:01.261275169Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.11.2 instance.id=c62adfe4-979f-44c9-a3a1-c4adb3f2c4fe service=registry version=v2.7.1
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: time="2023-06-29T09:19:01.261469049Z" level=info msg="redis not configured" go.version=go1.11.2 instance.id=c62adfe4-979f-44c9-a3a1-c4adb3f2c4fe service=registry version=v2.7.1
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: time="2023-06-29T09:19:01.261389383Z" level=info msg="Starting upload purge in 44m0s" go.version=go1.11.2 instance.id=c62adfe4-979f-44c9-a3a1-c4adb3f2c4fe service=registry version=v2.7.1
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: time="2023-06-29T09:19:01.27530762Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.11.2 instance.id=c62adfe4-979f-44c9-a3a1-c4adb3f2c4fe service=registry version=v2.7.1
Jun 29 09:19:01 ip-10-0-5-65 podman[1226]: time="2023-06-29T09:19:01.276878659Z" level=info msg="listening on [::]:5000, tls" go.version=go1.11.2 instance.id=c62adfe4-979f-44c9-a3a1-c4adb3f2c4fe service=registry version=v2.7.1
Jun 29 09:23:04 ip-10-0-5-65 podman[1226]: 54.80.58.62 - - [29/Jun/2023:09:23:04 +0000] "HEAD / HTTP/2.0" 200 0 "" "curl/7.61.1"
+ grep 'Active: active (running) since' /logs/artifacts/poc-registry-5000.status
     Active: active (running) since Thu 2023-06-29 09:18:54 UTC; 4min 12s ago
+ echo Succeed.
Succeed.
+ return 0
```

**Comment by yunjiang29** [comment]
@jianlinliu @gpei PTAL, it's ready for merge

File: ci-operator/step-registry/cucushift/installer/check/mirror-registry-services/cucushift-installer-check-mirror-registry-services-ref.metadata.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+	"path": "cucushift/installer/check/mirror-registry-services/cucushift-installer-check-mirror-registry-services-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/mirror-registry-services/cucushift-installer-check-mirror-registry-services-ref.metadata.json:2]: How about moving the step to `ci-operator/step-registry/mirror-images/` ?

Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/check/mirror-registry-services/cucushift-installer-check-mirror-registry-services-ref.metadata.json]: Moved to `mirror-images-check-registry-service`

**Pull request merged**